### PR TITLE
Zambia: optimizations, added special holidays

### PR DESCRIPTION
**Pull request by KJhellico**

### CODE REVIEW

File: holidays/countries/zambia.py [hunk 1]
```diff
@@ -27,70 +28,77 @@ class Zambia(HolidayBase):
...
+            (JUL, 7, "Funeral of Kenneth Kaunda"),
+            (AUG, 12, "General elections"),
+            (AUG, 13, "Counting in general elections"),
+            (AUG, 24, "Presidential inauguration"),
+        ),
+        2022: ((MAR, 18, "Funeral of Rupiah Banda"),),
+    }
 
     def _populate(self, year):
         super()._populate(year)
```
Comment (arkid15r) [inline holidays/countries/zambia.py:56]: I'd move it below the line 60.

**Review by arkid15r — COMMENTED** [review]
LGTM

File: holidays/countries/zambia.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/zambia.py:56]: > I'd move it below the line 60.

`special_holidays` declaration?

Comment (arkid15r) [reply holidays/countries/zambia.py:56]: Nope :).
I see the weird code selection here too, my bad. 

I meant `super()._populate(year)` line. It should be executed after the year condition check is satisfied.

Comment (KJhellico) [reply holidays/countries/zambia.py:56]: Done.

**Comment by dr-prodigy** [comment]
Thank you! KJ 👍

**Pull request merged**